lib/admin/WP_Auth0_Admin_Advanced.php

```diff
@@ -134,294 +134,233 @@ public function render_passwordless_method() {
+        $this->get_dashboard_link()
+      );
 
-  public function render_migration_ws() {
-    $v = $this->options->get( 'migration_ws' );
-    $token = $this->options->get( 'migration_token' );
+      $this->render_field_description( 'Security token:' );
 
-    echo $this->render_a0_switch( "wpa0_auth0_migration_ws", "migration_ws", 1, 1 == $v );
+      printf(
+        '<textarea class="code" rows="%d" disabled>%s</textarea>',
+        $this->_textarea_rows,
+        sanitize_text_field(  $this->options->get( 'migration_token' ) )
+      );
 
-    if ( $v ) {
-?>
-      <div class="subelement">
-        <span class="description"><?php echo __( 'Users migration is enabled. If you disable this setting, it can not be automatically enabled again, it needs to be done manually in the Auth0 dashboard.', 'wp-auth0' ); ?></span>
-        <br>
-        <span class="description"><?php echo __( 'Security token:', 'wp-auth0' ); ?></span>
-        <textarea class="code" disabled><?php echo $token; ?></textarea>
-      </div>
-    <?php
     } else {
-?>
-      <div class="subelement">
-        <span class="description"><?php echo __( 'Users migration is disabled. Enabling it will expose the migration webservices but the connection needs to be updated manually on the Auth0 dashboard. More info about the migration process ', 'wp-auth0' ); ?><a target="_blank" href="https://auth0.com/docs/connections/database/migrating">HERE</a>.</span>
-      </div>
-    <?php
+      $this->render_field_description(
+        __( 'Users migration is disabled. ', 'wp-auth0' ) .
+        __( 'Enabling this exposes migration webservices but the Connection must be updated manually. ', 'wp-auth0' ) .
+        $this->get_docs_link( 'users/migrations/automatic', __( 'More information here', 'wp-auth0' ) )
+      );
     }
-
   }
 
-  public function render_migration_ws_ips_filter() {
-    $v = $this->options->get( 'migration_ips_filter' );
-    $this->render_a0_switch( "wpa0_auth0_migration_ips_filter", "migration_ips_filter", 1, 1 == $v );
+  public function render_migration_ws_ips_filter( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
   }
 
-  public function render_migration_ws_ips() {
-	  $list = $this->options->get( 'migration_ips' );
-	  ?>
-	  <div class="subelement">
-		  <textarea name="<?php echo $this->options->get_options_name(); ?>[migration_ips]" id="wpa0_auth0_migration_ips"><?php echo $list; ?></textarea>
-		  <span class="description">
-	        <?php echo __( 'Only requests from this IPs will be allowed to the migration WS.', 'wp-auth0' ); ?>
-	        <?php echo __( 'Separate multiple IPs with commas.', 'wp-auth0' ); ?>
-        </span>
-	  </div>
-	  <?php
+  public function render_migration_ws_ips( $args ) {
+    $this->render_textarea_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Only requests from this IPs will be allowed to the migration WS. ', 'wp-auth0' ) .
```

Comment:
"will be allowed to the migration WS." I don't understand this one. Is WS webserver? Don't feel bad for writing the large version.